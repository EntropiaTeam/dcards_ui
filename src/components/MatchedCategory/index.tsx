import React, { FC } from 'react';
import { TypographyClassKey, Grid } from '@material-ui/core';
import { Category } from '../../context/categories/types/CategoriesTypes';
import TranslatedTypography from '../UI/TranslatedTypography';
import CategoryPreview from '../CategoryPreview';

interface MatchedCategoryProps {
  className: string;
  classes: Partial<Record<TypographyClassKey, string>>;
  matchedCategory: Category[];
}

const MatchedCategory: FC<MatchedCategoryProps> = ({ className, classes, matchedCategory }) => (
  <Grid container className={className}>
    <TranslatedTypography classes={classes} i18nKey="categoriesList.matchedTitle" />
    <Grid container>
      {
        matchedCategory.map((category: Category) => (
          <CategoryPreview key={`category-${category.name}-${Date.now()}`} category={category} />
        ))
      }
    </Grid>
  </Grid>

);

export default MatchedCategory;
