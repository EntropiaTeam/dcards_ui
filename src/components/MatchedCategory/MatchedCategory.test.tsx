import React from 'react';
import renderer from 'react-test-renderer';
import MatchedCategory from '.';

interface IntlMock {
  useTranslation: () => void;
}

interface TranslationMock {
  t: () => void;
}

jest.mock('react-i18next', (): IntlMock => ({
  useTranslation: (): TranslationMock => ({
    t: (key: string): string => key
  })
}));

describe('<MatchedCategory />', () => {
  const props = {
    matchedCategory: [{
      name: 'Test category',
      cards: [],
      preview_cards: [{}]
    }]
  };

  const MatchedCategoryComponent = renderer.create(<MatchedCategory {...props} />).toJSON();

  it('should render MatchedCategory component correctly', () => {
    expect(MatchedCategoryComponent).toMatchSnapshot();
  });
});
