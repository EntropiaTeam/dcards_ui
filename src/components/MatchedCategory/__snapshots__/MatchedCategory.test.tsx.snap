// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MatchedCategory /> should render MatchedCategory component correctly 1`] = `
<div
  className="MuiGrid-root MuiGrid-container"
>
  <p
    className="MuiTypography-root MuiTypography-body1"
  >
    categoriesList.matchedTitle
  </p>
  <div
    className="MuiGrid-root MuiGrid-container"
  >
    <div
      className="MuiGrid-root makeStyles-categoryContainer-1 MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-6 MuiGrid-grid-md-3 MuiGrid-grid-lg-3 MuiGrid-grid-xl-3"
      onClick={[Function]}
      onKeyDown={[Function]}
    >
      <div
        className="makeStyles-imagesContainer-2"
      >
        <div
          className="MuiBox-root MuiBox-root-6 makeStyles-imageContainer-4 makeStyles-imageContainer-5 makeStyles-imgContainer-3"
          tabIndex={-1}
        >
          <img
            alt={Object {}}
            className="makeStyles-image-7 makeStyles-image-8"
            src="/assets/i/[object Object]_thumb_front.png"
          />
        </div>
      </div>
      <p
        className="MuiTypography-root MuiTypography-body1"
      >
        Test category
      </p>
    </div>
  </div>
</div>
`;
